src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -7789,35 +7789,33 @@ public static W MultiplyAddWidening<W, N>(W op1, N op2, N op3)
             where W : IBinaryInteger<W>
             where N : IBinaryInteger<N>
         {
-            dynamic a = op2;
-            dynamic b = op3;
-            W product = (W)((W)a * (W)b);
-            W r = (W)(op1 + product);
-            return r;
+            W a = W.CreateChecked(op2);
+            W b = W.CreateChecked(op3);
+            W product = W.CreateTruncating(a * b);
+            return W.CreateTruncating(op1 + product);
         }
 
         public static W MultiplySubtractWidening<W, N>(W op1, N op2, N op3)
             where W : IBinaryInteger<W>
             where N : IBinaryInteger<N>
         {
-            dynamic a = op2;
-            dynamic b = op3;
-            W product = (W)((W)a * (W)b);
-            W r = (W)(op1 - product);
-            return r;
+            W a = W.CreateChecked(op2);
+            W b = W.CreateChecked(op3);
+            W product = W.CreateTruncating(a * b);
+            return W.CreateTruncating(op1 - product);
         }
 
         public static N AddRoundedHighNarrowing<W, N>(W op1, W op2)
             where W : IBinaryInteger<W>
             where N : IBinaryInteger<N>
         {
             int halfsize = default(N).GetByteCount() * 8;
-            dynamic a = op1;
-            dynamic b = op2;
-            ulong sum = (ulong)a + (ulong)b;
+            ulong a = Convert.ToUInt64(op1);
+            ulong b = Convert.ToUInt64(op2);
```

Comment:
These should be `ulong.CreateTruncating(op1)` or similar instead.

`Convert.*` is largely an expensive legacy API that isn't recommended for use when the `INumber<T>.Create` APIs are available.